include/uk/arch/spinlock.h

```diff
@@ -21,20 +23,62 @@
  * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
  * SUCH DAMAGE.
  */
-/* Taken from Mini-OS */
 
 #ifndef __UKARCH_SPINLOCK_H__
 #define __UKARCH_SPINLOCK_H__
 
+#include <uk/config.h>
+#include <stdint.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-typedef struct {} spinlock_t;
+#ifdef CONFIG_UKPLAT_LCPU_MULTICORE
+/** spinlock struct */
+typedef struct {
+	volatile uint32_t lock;
+} __spinlock_t;
```

Comment:
As this file will be part of the core API, it should not be dependent on `stdint.h`. You can use a simple `int`. The `include` statement can then be removed. For the basic data types, we also would like to remove the `_t` just as `__atomic`.
```suggestion
typedef struct {
	volatile int lock;
} __spinlock;
```